CMakeLists.txt

```diff
@@ -0,0 +1,213 @@
+cmake_minimum_required (VERSION 3.0.2)
+project (cjose)
+
+include(CheckSymbolExists)
+include(CheckIncludeFile)
+include(CheckTypeSize) 
+
+option(ENABLE_Tests "Turn on Tests" ON)
+option(CJOSE_BUILD_SHARED_LIBS "Build shared libraries." OFF)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/CMakeModules")
+
+# Build Release by default
+if (NOT CMAKE_BUILD_TYPE)
+set(CMAKE_BUILD_TYPE Release)
+else()
+endif()
+
+# Setup cjose version
+set(CJOSE_MAJOR_VERSION 0)
+set(CJOSE_MINOR_VERSION 4)
+set(CJOSE_PATCH_VERSION 1)
+set(PACKAGE_VERSION ${CJOSE_MAJOR_VERSION}.${CJOSE_MINOR_VERSION}.${CJOSE_PATCH_VERSION})
+  
+# Set output directories
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
+
+# Create version.h file
+configure_file (
+  "${PROJECT_SOURCE_DIR}/include/cjose/version.h.in"
+  "${PROJECT_BINARY_DIR}/include/cjose/version.h"
+  )
+
+# Set include directories  
+include_directories(
+${CMAKE_CURRENT_SOURCE_DIR}/include
+${CMAKE_CURRENT_SOURCE_DIR}/src/include
+${CMAKE_CURRENT_SOURCE_DIR}/src
+)
+
+# Find dependencies
+# Look for Homebrew version of OpenSSL
+function(GetRootDirForHomebrew PKG_NAME)
+    if ((NOT ${PKG_NAME}_INCLUDE_DIR) OR (NOT ${PKG_NAME}_LIBRARIES))
+        string(TOLOWER ${PKG_NAME} PKG_NAME_LOWER) 
+        file(GLOB HB_${PKG_NAME}_ROOT_DIRECTORIES /usr/local/Cellar/${PKG_NAME_LOWER}/*)
+        list(GET HB_${PKG_NAME}_ROOT_DIRECTORIES 0 HB_${PKG_NAME}_ROOT_DIR)
+        if(HB_${PKG_NAME}_ROOT_DIRECTORIES AND NOT ${PKG_NAME}_ROOT_DIR)
+            SET(${PKG_NAME}_ROOT_DIR ${HB_${PKG_NAME}_ROOT_DIR} PARENT_SCOPE)
+        endif(HB_${PKG_NAME}_ROOT_DIRECTORIES AND NOT ${PKG_NAME}_ROOT_DIR)
+    endif((NOT ${PKG_NAME}_INCLUDE_DIR) OR (NOT ${PKG_NAME}_LIBRARIES))
+endfunction(GetRootDirForHomebrew PKG_NAME)
+
+if(CMAKE_C_COMPILER_ID MATCHES "AppleClang")

```

Comment:
Add priority to homebrew versions of openssl, jansson and check. We can always specify custom locations of those packages by passing OPENSSL_ROOT_DIR, JANSSON_ROOT_DIR and CHECK_ROOT_DIR to our cmake invocation.